### PR TITLE
Add `day_of_week` and `day_of_year` to Column and DB_Column

### PR DESCRIPTION
**Pull request by jdunkerley**
### Pull Request Description

- Adds support for getting the weekday as an integer (1 Monday - 7 Sunday - ISO standard).
- Add support for getting the day of year.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

File: test/Table_Tests/src/Common_Table_Operations/Date_Time_Spec.enso [hunk 1]
```diff
@@ -84,13 +84,15 @@ add_specs suite_builder setup =
...
-        group_builder.specify "should allow to get the year/month/day of a Date" <|
+        group_builder.specify "should allow to get the year/month/day/day_of_year/day_of_week of a Date" <|
             t = data.dates
             a = t.at "A"
             a.year . to_vector . should_equal [2020, 2024, 1990, Nothing]
             a.month . to_vector . should_equal [12, 2, 1, Nothing]
             a.day . to_vector . should_equal [31, 29, 1, Nothing]
-            [a.year, a.month, a.day].each c->
+            a.day_of_year . to_vector . should_equal [366, 60, 1, Nothing]
+            a.day_of_week . to_vector . should_equal [4, 4, 1, Nothing]
```
Comment (radeusgd) [inline test/Table_Tests/src/Common_Table_Operations/Date_Time_Spec.enso]: Ideally would be good to have a test covering Sunday as well to ensure it works as advertised in all backends (as I can easily imagine some SQL backend doing it wrong by default)

**Review by radeusgd — APPROVED** [review]

File: test/Table_Tests/src/Common_Table_Operations/Date_Time_Spec.enso [hunk 2]
```diff
@@ -99,13 +101,14 @@ add_specs suite_builder setup =
             ((a.month) == (t.at "X")).to_vector . should_equal [False, False, True, Nothing]
             ((a.day) == (t.at "X")).to_vector . should_equal [False, True, True, Nothing]
 
-        group_builder.specify "should allow to get the year/month/day of a Date_Time" <|
+        group_builder.specify "should allow to get the year/month/day/day_of_year/day_of_week of a Date_Time" <|
             t = data.datetimes
             a = t.at "A"
             a.year . to_vector . should_equal [2020, 2024, 1990, Nothing]
             a.month . to_vector . should_equal [12, 2, 1, Nothing]
-            a.day . to_vector . should_equal [31, 29, 1, Nothing]
```
Comment (AdRiley) [inline test/Table_Tests/src/Common_Table_Operations/Date_Time_Spec.enso:107]: Don't think you meant to delete the day test

Comment (jdunkerley) [reply test/Table_Tests/src/Common_Table_Operations/Date_Time_Spec.enso:107]: No didn't mean to remove that.

File: test/Table_Tests/src/Common_Table_Operations/Date_Time_Spec.enso [hunk 3]
```diff
@@ -99,13 +101,14 @@ add_specs suite_builder setup =
...
-            [a.year, a.month, a.day].each c->
+            a.day_of_year . to_vector . should_equal [366, 60, 1, Nothing]
```
Comment (jdunkerley) [inline test/Table_Tests/src/Common_Table_Operations/Date_Time_Spec.enso]:
```suggestion
            a.day . to_vector . should_equal [31, 29, 1, Nothing]
            a.day_of_year . to_vector . should_equal [366, 60, 1, Nothing]
```

**Review by AdRiley — APPROVED** [review]

**Pull request merged**